File: react-java/backend/src/main/java/com/clt/hrm/core/administration/user/dtos/UserInfoDto.java
package com.clt.hrm.core.administration.user.dtos;

import com.clt.hrm.infra.common.dtos.BaseDto;
import com.clt.hrm.infra.file.dtos.FileDto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
public class UserInfoDto extends BaseDto {
	private static final long serialVersionUID = 3244721131774333588L;
	private String usrId;
	private String usrPwd;
    private String usrNm;
    private String usrEml;
    private String roleId;
    private String roleNm;
    private String langVal;
    private String sysModVal;
    private String dtFmtVal;
    private String sysColrVal;
	private String usrPhn;
	private String usrAddr;
	private String usrDesc;
	private String usrFileId;
	private FileDto usrFileDto;
}

File: nestjs/backend/src/apps/administration/users/dto/user-info.dto.ts
import {
  IsEmail,
  IsOptional,
  IsString,
  MaxLength,
} from 'class-validator';
import { BaseDto } from '@infra/common/dto';

export class UserInfoDto extends BaseDto {
  @IsOptional()
  @IsString()
  usrId?: string;

  @IsOptional()
  @IsString()
  @MaxLength(200)
  usrNm?: string;

  @IsOptional()
  @IsString()
  usrPwd?: string;

  @IsOptional()
  @IsEmail()
  usrEml?: string;

  @IsOptional()
  @IsString()
  @MaxLength(20)
  usrPhn?: string;

  @IsOptional()
  @IsString()
  @MaxLength(500)
  usrAddr?: string;

  @IsOptional()
  @IsString()
  usrDesc?: string;

  @IsOptional()
  @IsString()
  usrFileId?: string;

  @IsOptional()
  @IsString()
  roleId?: string;

  @IsOptional()
  @IsString()
  roleNm?: string;

  @IsOptional()
  @IsString()
  langVal?: string;

  @IsOptional()
  @IsString()
  sysModVal?: string;

  @IsOptional()
  @IsString()
  dtFmtVal?: string;

  @IsOptional()
  @IsString()
  sysColrVal?: string;
}
